module/VuFind/src/VuFind/View/Helper/Root/Record.php

```diff
@@ -570,9 +593,35 @@ public function getQrCode($context, $extra = [], $level = "L", $size = 3,
      */
     public function getThumbnail($size = 'small')
     {
-        return $this->coverRouter
-            ? $this->coverRouter->getUrl($this->driver, $size)
-            : false;
+        $handlers = $this->coverLoader->getHandlers();
```

Comment:
This refactoring seems to be moving a lot of work from the cover router into the record view helper. Might it be preferable to pass additional parameters into the router to be able to do the work there? (I haven't reviewed this closely, so maybe there is a strong reason to do it this way... but in general, my preference would be to keep the view helper thinner and delegate work to a more specialized class, if practical).

Reply:
I moved this logic to cover router, it does make more sense here. Thanks.